app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      flash[:error] = "Could not find task with id: #{params[:id]}"
+      redirect_to tasks_path, status: 302
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    task = Task.new(task_params) # instantiate a new task
+    if task.save # save returns true if the database insert succeeds
+      redirect_to task_path(task.id) # go to the index so we can see the task in the list
+    else # save failed :(
+      flash[:error] = "Save was unsuccessful. Try again!"
+      render :show # show the new task form view again
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      flash[:error] = "Could not find task with id: #{params[:id]}"
+      redirect_to tasks_path, status: 302
+    end
+  end
+
+  def update
+    task = Task.find_by(id: params[:id])
+
+    if task.nil?
+      flash[:error] = "Could not find task with id: #{params[:id]}"
+      redirect_to tasks_path
+    else task.update_attributes(task_params)
```

Comment:
for pure syntax/readability reasons, it'll be best to bump the code off of the line with the `else`, so it's

```ruby
else
  task.update_attributes(task_params)
```